lib/request-helpers.js

```diff
@@ -0,0 +1,27 @@
+/**
+ *
+ * @param {import('./typings.js').RequestWithSubnet} request
+ * @param {import('fastify').FastifyReply} reply
+ */
+export const subnetPreHandlerHook = async (request, reply) => {
+  // @ts-ignore TODO: Fix
+  const { subnet } = request.params
+  if (!subnet || typeof subnet !== 'string') {
+    reply.code(400).send({ error: 'Invalid subnet parameter' })
+    return
+  }
+
+  if (!isValidSubnet(subnet.toLowerCase())) {
+    reply.code(404).send()
+    console.log('Invalid subnet parameter')
+  }
+}
+
+/**
+ * Validate if the given subnet is either 'arweave' or 'walrus'
+ * @param {string} subnet
+ * @returns {boolean}
```

Comment:
If you write something like the following (I don't remember the exact TS syntax), then TypeScript will be able to understand that after calling `isValidSubnet`, the `subnet` variable is no longer "arbitrary string" but must be `"walrus" | "arweave"`.

```suggestion
 * @returns {subnet is Subnet}
```

Reply:
You can learn more about user-defined type-guards here: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
Amazing! Didn't know about this feature. Thanks for sharing! 🙌🏻